### PR TITLE
Courier may complain about the method only

### PR DESCRIPTION
**Pull request by szepeviktor**
> Mar 30 22:29:18 szerver imapd-ssl: LOGIN FAILED, method=PLAIN, ip=[::ffff:1.2.3.4]

### CODE REVIEW

File: config/filter.d/courier-auth.conf [hunk 1]
```diff
@@ -11,7 +11,7 @@ before = common.conf
 
 _daemon = (?:courier)?(?:imapd?|pop3d?)(?:login)?(?:-ssl)?
 
-failregex = ^%(__prefix_line)sLOGIN FAILED, user=.*, ip=\[<HOST>\]$
+failregex = ^%(__prefix_line)sLOGIN FAILED, (user|method)=.*, ip=\[<HOST>\]$
```
Comment (sebres) [inline config/filter.d/courier-auth.conf]: please non-captured groups:
```diff
- (user|method)
+(?:user|method)
```

Comment (szepeviktor) [inline config/filter.d/courier-auth.conf]: Done. Please squash the two commits.

**Comment by sebres** [comment]
Don't forget the tests and changelog :)

**Comment by szepeviktor** [comment]
I've forgot how to add a test.
How to deal with the date?
```
# failJSON: { "time": "2005-04-17T19:17:11", "match": true , "host": "1.2.3.4" }
Apr 17 19:17:11 SERVER courierpop3login: LOGIN FAILED, user=USER@EXAMPLE.org, ip=[::ffff:1.2.3.4]
```

**Comment by sebres** [comment]
Our test cases running in CET in year 2004/2005...
Just use the same date as in test cases above + 1 second (and repeat it in JSON).
```
# failJSON: { "time": "2005-04-17T19:17:12", "match": true , "host": "192.0.2.4" }
Apr 17 19:17:12 server imapd-ssl: LOGIN FAILED, method=PLAIN, ip=[::ffff:192.0.2.4]
```
BTW. And I don't like usage of "real" IP (like 1.2.3.4 etc.). There are special test-IPs assigned as "TEST-NET" - [192.0.2.1/24](https://en.wikipedia.org/wiki/Reserved_IP_addresses).

**Comment by szepeviktor** [comment]
Thank you.

**Comment by sebres** [comment]
Looks good. 
I would merge it, but still one question: Are you sure, that this will be not occurred as preauth-step by normal connects (handshake resp. initial connect without user name)?

**Comment by szepeviktor** [comment]
AFAIK a legitimate client will not connect without credentials.

The only thing I am sure of is attacks happen this way.

**Comment by sebres** [comment]
> AFAIK a legitimate clients will not connect without credentials.

Well, no courier at all. I asked a friend (having courier) to do (miss-)connect to his mail-server and give me the log-excerpt...
Here you are:
```
16:30:57 srv imapd: LOGIN FAILED, method=PLAIN, ip=[::ffff:xxx.xxx.xxx.xxx]
16:31:02 srv imapd: LOGIN FAILED, user=test, ip=[::ffff:xxx.xxx.xxx.xxx]
16:31:11 srv imapd: Disconnected, ip=[::ffff:xxx.xxx.xxx.xxx], time=14, starttls=1
16:34:54 srv imapd: Connection, ip=[::ffff:xxx.xxx.xxx.xxx]
16:34:55 srv imapd: LOGIN FAILED, method=PLAIN, ip=[::ffff:xxx.xxx.xxx.xxx]
16:35:00 srv imapd: LOGIN FAILED, user=test, ip=[::ffff:xxx.xxx.xxx.xxx] 
```

Either it does this always (additionally to `LOGIN FAILED, user=`) and then we'll have 2 failures by exactly 1 attempt.
Or the method is indeed disabled (but he says - no, PLAIN is enabled)
Or I miss something here.

As long it is not clear, I cannot merge it (or we should provide this optional, similar `sshd[mode=aggressive]`)...

BTW. He says that he cannot reproduce a method-only entry (without user). Version conditionally?
Can you try to reproduce such situation?

**Comment by szepeviktor** [comment]
This happens on my end

```
May  2 12:21:17 web imapd-ssl: Connection, ip=[::ffff:61.180.34.44]
May  2 12:21:27 web imapd-ssl: LOGIN FAILED, method=PLAIN, ip=[::ffff:61.180.34.44]
```
whois netname:        CHINANET-JX

**Pull request merged**